### ISSUE TITLE
Plugins client side

### ISSUE THREAD

**Issue by lawrencenull** [spencermountain/wtf_wikipedia#421]
I was looking to prototype something in jsfiddle and realized there isn't really any documentation on setting up or running the plugins client side, or links to them on unpkg.com. It would probably be good to create a fiddle with all the plugins loaded as an example and that someone could simply fork when they want to prototype or test something.

**Comment by spencermountain** [comment]
hey @lawrencenull great idea -  I'll admit I haven't kept-up with the codepen/jsfiddle/Glitch scene. 
I agree the client-side use is opaque - I'd love some help with this.
cheers

**Comment by lawrencenull** [comment]
If you have the plugins already on unpkg.com and set up a jsfiddle including them I could build out a few use case examples.

**Comment by spencermountain** [comment]
ah cool - yep, they should be there -
```html
<script src="https://unpkg.com/compromise-numbers" />
<script>
console.log(compromiseNumbers)
</script>
```
They should all export a camel-cased variable globally.
Let me know if I can help.

**Comment by lawrencenull** [comment]
It seems to be failing loading plugins without the use of require.

example:  wtf.extend('wtf-plugin-classify');
console.log('wtf-plugin-classify');
wtf.fetch('Toronto Raptors').then((doc) => {
  let res = doc.classify();
  console.log(res);
})

Just throws a script error. I tried with two different plugins, either I'm doing something wrong or it's with the WTF library.

https://jsfiddle.net/lawrencenull/wzrsh06k/3/

**Comment by spencermountain** [comment]
hey Lawrence, sorry about this confusion - You're right that this isn't clear. 
you've got to pass the actual plugin into `wtf.extend()` and not a string.
```html
<script src="https://unpkg.com/wtf_wikipedia" />
<script src="https://unpkg.com/wtf-plugin-classify" />
<script>
wtf.extend(wtfClassify)
wtf.fetch('Toronto Raptors').then((doc) => {
  let res = doc.classify();
  console.log(res);
})
</script>
```

depending on your situation, you can add `async` tags to the scripts you want to wait for, or wrap `wtf.extend()` etc after a body.onLoad call, or what-have-you.

This documentation should really improve. You can see most plugins have a running example on observable notebook, but maybe we should commit a basic demo.html examples too.

PR welcome!
cheers

**Comment by lawrencenull** [comment]
Runs perfectly now. I'll make a few fiddles for examples and try to document them well.

**Comment by lawrencenull** [comment]
A quick client-side example in JsFiddle https://jsfiddle.net/5pwr2m84/2/.

**Comment by spencermountain** [comment]
nice one!

**Comment by spencermountain** [comment]
want me to put a link on the readme?

**Comment by lawrencenull** [comment]
If you'd like, then someone could just fork it and change the plugins to whatever they need.

**Comment by lawrencenull** [comment]
Here is a much better example using Classify, Summary, and HTML plugins. It uses case breaks to split the classification and adds a profile image from Deezer for artists and MusicGroups and renders a google map for places. I was going to add TMDB for actors but I'd have to use my API key. The case breaks can of course be extended to all types and interact with any outside or internal WTF API.

https://codepen.io/lawrencenull/pen/zYooawM

**Comment by spencermountain** [comment]
hey that's great!

**Comment by lawrencenull** [comment]
I did want to make you aware that for some reason classify returns Empire State Building back as a person. Not really sure why.

**Comment by lawrencenull** [comment]
A second plugin and wft integration. An autocomplete that directly calls wtf with the pageid of the search api call. The demo is set up to render the first section, however not all articles have this, I'll probably switch this to call wiki summary's api since almost all articles do have a summary. That kind of defeats the point of the demo though.

demo: https://codepen.io/lawrencenull/full/YzpjMYN
required autocomplete js: https://unpkg.com/@trevoreyre/autocomplete-js
required autocomplete css: https://unpkg.com/@trevoreyre/autocomplete-js/dist/style.css

Plan on adding some wikidata integration to create a complete infobox and a basic starter template to render a full site if anyone is interested or needs it.

**Comment by spencermountain** [comment]
hey! that's really nice @lawrencenull 
I'm gonna put a link to these on the readme, is that okay?

**Comment by lawrencenull** [comment]
Of course, doing them to help out anyone using the library. Working on an infobox and complete template.

**Comment by MarketingPip** [comment]
Tho this is old and should be closed. You can also use this to import your plugins. 

```html
<script type="module">
import wtf from "wtf-wikipedia"
import * as nameOfPlugin from "plugin-name";
wtf.extend( nameOfPlugin)
</script>
```

@spencermountain - I was thinking about maybe a PR the other day to state this but! Did not know if it was needed. I would assume it would be more than useful - seeing this issue & running into some confusion for a brief moment the other day on my end.

**Comment by spencermountain** [comment]
hey! apologies for the delay on this - I've been off-keyboard for a couple weeks.
Yes - any help documenting the weird corners is always welcomed - I don't understand most of it, myself.
Thank you!
